### ISSUE TITLE
Improve text rendering performance with custom fonts in reviewer.

### ISSUE THREAD

**Issue by brishtibheja** [ankidroid/Anki-Android#15938]
I'm using a font called Klee One on my Japanese cards. As it currently stands, my review time is increased because It takes some time for the text on the Back of a card to load. I can only answer after I've seen the back of the card. I would appreciate if this is improved. This is probably not device specific issue. I saw this on my Xiaomi device as well as my Realme one. This also happens with other jp fonts.

**Comment by BrayanDSO** [comment]
I believe that you are using 2.17 (you should've added the debug info as the issue template requests), so I'm going to ask you to do a test to see the path to improve that.

1. Enable developer options (here's a [link](https://github.com/ankidroid/Anki-Android/issues/13878))
2. Enable the `New previewer` setting (it won't be dangerous and already is the default for 2.18)
3. Check if the performance is better in the previewer

In a nutshell, I'm asking to check if the "new rendering mechanism" used at the new previewer helps in your issue, since the reviewer should start using it in the future.

**Comment by BrayanDSO** [comment]
Also, https://github.com/ankidroid/Anki-Android/issues/15863#issuecomment-1992190692 would help, since fonts aren't loaded with `file:///` by default anymore, which is significantly faster

**Comment by 2aecfff4** [comment]
@BrayanDSO 
> 2\. Enable the `New previewer` setting (it won't be dangerous and already is the default for 2.18)

This option was unintentionally removed in https://github.com/ankidroid/Anki-Android/commit/61cd5d0f0b8b2e98ec01d151dcc6b078bca8fc37

**Comment by david-allison** [comment]
@2aecfff4 Everything but the dev option was removed in 35f475ff97c5ceaaef45462094bcd826ffa89eed

The old previewer was removed and there was no reason to keep the setting

**Comment by david-allison** [comment]
@mikehardy FYI: minor issue with 2.17.6 removing the 'New Previewer' development-only setting

**Comment by brishtibheja** [comment]
@BrayanDSO I was using 2.18 alpha versions until recently then my new phone got stolen and now I'm using 2.17

I didn't include the ver. info because I have seen this since 2.15 so didn't thought it would be necessary.

> if the "new rendering mechanism" used at the new previewer helps in your issue

sadly no

**Comment by brishtibheja** [comment]
> Also, [#15863 (comment)](https://github.com/ankidroid/Anki-Android/issues/15863#issuecomment-1992190692) would help, since fonts aren't loaded with `file:///` by default anymore, which is significantly faster

The only word I recognize there is CORS because some dev in Github once mentioned it. Sorry I'm actually tech illiterate. I'm in med.

**Comment by BrayanDSO** [comment]
> The only word I recognize there is CORS because some dev in Github once mentioned it. Sorry I'm actually tech illiterate. I'm in med.

That was a comment for other devs. Not actionable from your part

---


Using the same instance of the Webview, and therefore using cached content by default, (i.e. new rendering mechanism) + using the `file:` scheme are probably the only actionable stuff that we can do.

Besides that, it will be either a problem with the Android Webview, which is somewhat frequent for fonts, or a limitation of your phone.

Since all the actionable tasks that we can do are tracked somewhere else, I'll close this. But thanks for the report.

**Comment by brishtibheja** [comment]
@BrayanDSO Wanted to add that I imported a deck where this problem didn't occur. Apparently the font it used had a smaller file size. So I tried installing [this](https://github.com/fontworks-fonts/Klee/tree/master/fonts/ttf) which is 8 MB compared to my 19 MB file before. I think itthe rendering time has improved? I'm not quite sure if it is the file size of the font or something in my note template.

**Comment by BrayanDSO** [comment]
What I expect is exactly that. That there is a delay because the font is reloaded in every card flip. I hope that that is fixed at least in the previewer when flipping cards
